app/services/api/v0_1/create_hesa_trainee_detail_service.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Api
+  module V01
+    class CreateHesaTraineeDetailService
+      include ServicePattern
+
+      def initialize(trainee:)
+        @trainee = trainee
+      end
+
+      def call
+        return true unless trainee.hesa_trainee_detail.nil?
+
+        hesa_trainee_detail = Hesa::TraineeDetail.build(trainee:)
+
+        attributes = extract_attributes_from_student_record
+        attributes.merge!(extract_attributes_from_metadatum_record(attributes))
+        hesa_trainee_detail.assign_attributes(attributes)
+
+        hesa_trainee_detail.save
+      end
+
+    private
+
+      attr_reader :trainee
+
+      def extract_attributes_from_metadatum_record(existing_attributes)
+        return {} if trainee.hesa_metadatum.nil?
+
+        attributes_for_extraction = %i[itt_aim itt_qualification_aim pg_apprenticeship_start_date]
+        attributes_for_extraction = attributes_for_extraction.map! { |attribute| attribute if existing_attributes[attribute].nil? }.compact
+
+        trainee.hesa_metadatum.slice(attributes_for_extraction)
+      end
+
+      def extract_attributes_from_student_record
+        return {} if trainee.hesa_students.empty?
+
+        attributes_for_extraction = %i[course_age_range itt_aim itt_qualification_aim fund_code pg_apprenticeship_start_date ni_number]
+
+        attributes = trainee.hesa_students.latest.slice(attributes_for_extraction)
+        attributes.merge!(funding_method: trainee.hesa_students.last.bursary_level)
+      end
+    end
```

Comment:
Are those all the attributes that we can fetch from those records? I can see for example `previous_last_name` is missing and can be fetched by `Hesa::Student#last_surname`and `Hesa::Metadatum#fundability` can be assigned to `fund_code` according to [Considerations](https://dfedigital.atlassian.net/wiki/spaces/Register/pages/5255790594/Updates+to+existing+trainees+who+are+on+multi-year+courses#Considerations)

Reply:
Are the following compatible?

```
course_year -> Hesa::Student#year_of_course
course_study_mode -> Hesa::Student#mode
hesa_disabilities -> Hesa::Student#disability1,2,3,4,5,6,7,8,9
```